src/test/java/com/google/ehub/servlets/DashboardServletTest.java

```diff
@@ -0,0 +1,129 @@
+
+  private static final int MAX_SEARCH_VALUE_CHARS = 150;
+
+  @Mock HttpServletRequest request;
+
+  @Mock HttpServletResponse response;
+
+  @Before
+  public void init() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void getRequestWithNullParams_NoContentGetsSent() throws IOException {
+    when(request.getParameter(SEARCH_VALUE_PARAMETER_KEY)).thenReturn(null);
+    when(request.getParameter(SORTING_DIRECTION_PARAMETER_KEY)).thenReturn(null);
+    when(response.getWriter()).thenReturn(printWriter);
+
+    servlet.doGet(request, response);
+
+    verify(response, never()).setContentType(JSON_CONTENT_TYPE);
+  }
+
+  @Test
+  public void getRequestWithEmptyParams_NoContentGetsSent() throws IOException {
+    when(request.getParameter(SEARCH_VALUE_PARAMETER_KEY)).thenReturn("");
+    when(request.getParameter(SORTING_DIRECTION_PARAMETER_KEY)).thenReturn("");
+    when(response.getWriter()).thenReturn(printWriter);
+
+    servlet.doGet(request, response);
+
+    verify(response, never()).setContentType(JSON_CONTENT_TYPE);
+  }
+
+  @Test
+  public void getRequestWithValidParams_ContentGetsSent() throws IOException {
+    when(request.getParameter(SEARCH_VALUE_PARAMETER_KEY)).thenReturn("Star Wars");
+    when(request.getParameter(SORTING_DIRECTION_PARAMETER_KEY))
+        .thenReturn(ASCENDING_PARAMETER_VALUE);
+    when(response.getWriter()).thenReturn(printWriter);
+
+    servlet.doGet(request, response);
+
+    verify(response).setContentType(JSON_CONTENT_TYPE);
+  }
+
+  @Test
+  public void getRequestWithInvalidSortingParam_NoContentGetsSent() throws IOException {
+    when(request.getParameter(SEARCH_VALUE_PARAMETER_KEY)).thenReturn("Star Wars");
+    when(request.getParameter(SORTING_DIRECTION_PARAMETER_KEY)).thenReturn("Invalid Sorting");
+    when(response.getWriter()).thenReturn(printWriter);
+
+    servlet.doGet(request, response);
+
+    verify(response, never()).setContentType(JSON_CONTENT_TYPE);
+  }
+
+  @Test
+  public void getRequestWithExceedingSearchValueLength_NoContentGetsSent() throws IOException {
+    char[] searchValueChars = new char[500];
+    Arrays.fill(searchValueChars, 'a');
+    String largeSearchValue = new String(searchValueChars);
+
+    when(request.getParameter(SEARCH_VALUE_PARAMETER_KEY)).thenReturn(largeSearchValue);
+    when(request.getParameter(SORTING_DIRECTION_PARAMETER_KEY)).thenReturn("Invalid Sorting");
+    when(response.getWriter()).thenReturn(printWriter);
+
+    servlet.doGet(request, response);
+
+    verify(response, never()).setContentType(JSON_CONTENT_TYPE);
+  }
+
+  @Test
+  public void getRequestWithMaximumValidSearchValueLength_ContentGetsSent() throws IOException {
+    char[] searchValueChars = new char[MAX_SEARCH_VALUE_CHARS];
+    Arrays.fill(searchValueChars, 'a');
+    String maxValidSearchValue = new String(searchValueChars);
```

Comment:
This can be moved to a helper method:
```
private static String getSearchValue(int length) {
  char[] searchValueChars = new char[length];
  Arrays.fill(searchValueChars, 'a');
  return new String(searchValueChars);
}
```

Reply:
I added this helper method to avoid repetition in last commit.